Samples/Emoji8/UWP/cs/Emoji8/Pages/EmotionPage.xaml.cs

```diff
@@ -0,0 +1,310 @@
+            deferral.Complete();
+        }
+
+        protected override async void OnNavigatedTo(NavigationEventArgs e)
+        {
+            Application.Current.Suspending += Current_Suspending;
+            Application.Current.Resuming += Current_Resuming;
+            _pageNavigatedToTime = DateTime.Now;
+            if (_isInitialized)
+            {
+                await CleanUpCameraAsync();
+            }
+            await InitializeScreenAsync();
+        }
+
+        private async Task InitializeScreenAsync()
+        {
+            CameraHelperResult result = await CameraService.Current.InitializeAsync();
+            if (result != CameraHelperResult.Success)
+            {
+                await MessageDialogService.Current.WriteMessage(result.ToString() + GameText.LOADER.GetString("CameraHelperResultFailed"));
+                return;
+            }
+
+            bool isModelLoaded = await IntelligenceService.Current.InitializeAsync();
+            if (!isModelLoaded)
+            {
+                await MessageDialogService.Current.WriteMessage(GameText.LOADER.GetString("ModelLoadedFailed"));
+                return;
+            }
+            IntelligenceService.Current.ScoreUpdated += UpdatePredictionRadialGaugeValue;
+
+            //need dummy current emoji placeholder to start
+            if (!_isPaused) CurrentEmojis._currentEmoji = null;
+
+            //maintain results emojis if app is suspended and resumed
+            if (!_isPaused) _resultsEmojis = new List<Emoji>();
+
+            //to avoid initial interval delay, make first UI update here
+            await UpdateNonPredictionBasedUIAsync();
+
+            _isInitialized = true;
+        }
+
+        public void UpdatePredictionRadialGaugeValue(object sender, EmotionPageGaugeScoreEventArgs e)
+        {
+            nextPercentage = (double)e.Score;
+        }
+
+        private void InitializeTimer()
+        {
+            countdownNum = Constants.EMOJI_DISPLAY_DURATION_IN_SECONDS * Constants.DIVISIBLE_FACTOR;
+            if (EmojiTimer == null) EmojiTimer = new DispatcherTimer();
+            EmojiTimer.Interval = TimeSpan.FromMilliseconds(Constants.EMOJI_COUNTDOWN_INTERVAL);
+            EmojiTimer.Tick += DispatcherTimer_Tick;
+        }
+
+        private async void DispatcherTimer_Tick(object sender, object e)
+        {
+            await UpdateNonPredictionBasedUIAsync();
+            UpdatePredictionRadialGaugeAesthetics();
+        }
+
+        private async Task UpdateNonPredictionBasedUIAsync()
+        {
+            if (countdownNum == 0)
+            {
+                await UpdateGameplayScreenAsync();
+            }
+            else
+            {
+                UpdateCountdownRing();
+            }
+
+            if (countdownNum % Constants.DIVISIBLE_FACTOR == 0)
+            {
+                //display pattern 5...4...3...2...1...5... [...] ...2...1...0
+                if ((numEmojisLeftToPerform > -1 && countdownNum != 0) || 
```

Comment:
I haven't read through all the details of this, but could numEmojisLeftToPerform be negative? By looking at the variable name it seems to me like it should never be negative :)